poly-commitment/src/lagrange_cache.rs

```diff
@@ -0,0 +1,201 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    fs::{self, File},
+    marker::PhantomData,
+    path::{Path, PathBuf},
+};
+
+use crate::{commitment::CommitmentCurve, PolyComm};
+use ark_poly::Radix2EvaluationDomain as D;
+use std::hash::{Hash, Hasher};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum LagrangeCache<G> {
+    Local(FileCache<G>),
+    Remote(GoogleCloudCache<G>),
+}
+
+impl<G> Default for LagrangeCache<G> {
+    fn default() -> Self {
+        LagrangeCache::Local(FileCache::new(PathBuf::from("/tmp/lagrange_cache")))
+    }
+}
+
+// Keep this enum private for type safety
+pub enum LagrangeCacheKey {
+    Local(PathBuf),
+    Remote(()),
+}
+
+impl<G: CommitmentCurve> LagrangeCacheTrait<G> for LagrangeCache<G> {
+    type CacheKey = LagrangeCacheKey;
+
+    fn lagrange_basis_cache_key(&self, g: &Vec<G>, domain: &D<G::ScalarField>) -> Self::CacheKey {
```

Comment:
did we also need h here?

Reply:
You shouldn't do, `h` isn't included in the lagrange basis